server/resp/src/main/java/org/infinispan/server/resp/Resp3Handler.java

```diff
@@ -63,11 +84,11 @@ protected CompletionStage<RespRequestHandler> actualHandleRequest(ChannelHandler
             bufferToWrite.writeByte('\r').writeByte('\n');
             break;
          case "SET":
-            return stageToReturn(cache.putAsync(arguments.get(0), arguments.get(1)), ctx, OK_BICONSUMER);
```

Comment:
I was originally thinking we could do something more streamlined for the regular set case and then put the rest in a function like the following:

```java
if (arguments.size() != 2) {
   return handleSetWithoutTwoArguments(arguments);
}
return stageToReturn(cache.putAsync(arguments.get(0), arguments.get(1)), ctx, OK_BICONSUMER);
```

Reply:
Updated.